ide/editor.completion/src/org/netbeans/spi/editor/completion/CompletionCollector.java

```diff
@@ -0,0 +1,565 @@
+            @NonNull
+            public Completion build() {
+                return new Completion(label, kind, tags, detail, documentation, preselect,
+                        sortText, filterText, insertText, insertTextFormat, textEdit,
+                        additionalTextEdits, commitCharacters, command, data);
+            }
+        }
+    }
+
+    public static enum Kind {
+
+        Text(1),
+	Method(2),
+	Function(3),
+	Constructor(4),
+	Field(5),
+	Variable(6),
+	Class(7),
+	Interface(8),
+	Module(9),
+	Property(10),
+	Unit(11),
+	Value(12),
+	Enum(13),
+	Keyword(14),
+	Snippet(15),
+	Color(16),
+	File(17),
+	Reference(18),
+	Folder(19),
+	EnumMember(20),
+	Constant(21),
+	Struct(22),
+	Event(23),
+	Operator(24),
+	TypeParameter(25);
+
+        private final int value;
+
+	private Kind(int value) {
+            this.value = value;
+	}
+
+	public int getValue() {
+	    return value;
+	}
+    }
+
+    public static enum Tag {
+
+        Deprecated(1);
+
+        private final int value;
+
+        private Tag(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+    }
+
+    public static enum TextFormat {
+
+        /**
+         * The primary text to be inserted is treated as a plain string.
+         */
+        PlainText(1),
+
+        /**
+         * The primary text to be inserted is treated as a snippet.
+         */
+        Snippet(2);
+
+        private final int value;
+
+        private TextFormat(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+    }
+
+    public static final class TextEdit {
+
+        private final int start;
+        private final int end;
+        private final String newText;
+
+        public TextEdit(int start, int end, @NonNull String newText) {
+            this.start = start;
+            this.end = end;
+            this.newText = newText;
+        }
+
+        /**
+	 * The start offset of the text document range to be manipulated. To insert
+	 * text into a document create edit where {@code startOffset == endOffset}.
+	 */
+        public int getStartOffset() {
+            return start;
+        }
+
+        /**
+	 * The end offset of the text document range to be manipulated. To insert
+	 * text into a document create edit where {@code startOffset == endOffset}.
+	 */
+        public int getEndOffset() {
+            return end;
+        }
+
+        /**
+         * The string to be inserted. For delete operations use an empty string.
+         */
+        @NonNull
+        public String getNewText() {
+            return newText;
+        }
+    }
+
+    public static final class Command {
+
+        private final String title;
+        private final String command;
+        private final List<Object> arguments;
+
+        public Command(@NonNull String title, @NonNull String command, @NullAllowed List<Object> arguments) {
```

Comment:
What can be the types of `arguments`? Any `Object`?

Reply:
@jlahoda is asking: What is semantics of `Command` in NetBeans?

Reply:
Documentation enanced.